common/src/main/java/com/wynntils/models/players/FriendsModel.java

```diff
@@ -109,6 +132,33 @@ public void onChatReceived(ChatMessageReceivedEvent event) {
                 return;
             }
         }
+
+        if (onlineMessageStatus == ListStatus.EXPECTING
+                && styledText.getMatcher(ONLINE_FRIENDS_HEADER).matches()) {
+            // List of online friends is sent in multiple messages
+            // When we detect the first message indicating the start of the friends list, we set a flag
+            onlineMessageStatus = ListStatus.PROCESSING;
+            onlineFriends.clear();
+            event.setCanceled(true);
+            return;
+        }
+        if (onlineMessageStatus == ListStatus.PROCESSING) {
+            // If this flag is set, the next messages should be the list of online friends
+            // But as soon as the matcher fails, we know we've reached the end of the list
+            Matcher onlineFriendMatcher = styledText.getMatcher(ONLINE_FRIEND);
+            if (onlineFriendMatcher.matches()) {
+                String username = onlineFriendMatcher.group(1);
+                int server = Integer.parseInt(onlineFriendMatcher.group(2));
+
+                onlineFriends.put(username, server);
+                WynntilsMod.info("Friend " + username + " is online on WC" + server);
+                event.setCanceled(true);
+                return;
+            } else {
+                WynntilsMod.postEvent(new FriendsEvent.OnlineListed());
```

Comment:
As I general rule, try to finish the internal processing before posting events. I don't think anything can go wrong in this case, but it is more robust if you reset the  onlineMessageStatus back to ListStatus.IDLE before sending out the event, in case the event handler calls back into this model etc.